test/winter_onboarding_2021/fleet_management_service/handlers/user_test.clj

```diff
@@ -123,3 +129,59 @@
               :style-class "alert alert-danger"
               :message "You are not logged in"}
              (:flash response))))))
+
+(deftest login-form
+  (testing "Should show the login form page if the user is not already logged-in"
+    (let [response (handler/login-form {})
+          html-page-login (html5 (layout/application {}
+                                                     "Login"
+                                                     (user-view/login-form)))]
+      (is (= 200 (:status response)))
+      (is (= html-page-login (:body response)))))
+  (testing "Should redirect to /users/dashboard if user is logged-in"
+    (let [user (user-db/create (factories/user))
+          response (handler/login-form {:user (select-keys user 
+                                                           [:users/id
+                                                            :users/name
+                                                            :users/role
+                                                            :users/email])})]
+      (is (= 302 (:status response)))
+      (is (= "/users/dashboard"  (get-in response [:headers "Location"]))))))
+
+(deftest signup-form
+  (testing "Should show the sign form page if the user is not already logged-in"
+    (let [response (handler/signup-form{})
+          html-page-login (html5 (layout/application {}
+                                                     "Sign up"
+                                                     (user-view/signup-form)))]
+      (is (= 200 (:status response)))
+      (is (= html-page-login (:body response)))))
+  (testing "Should redirect to /users/dashboard if user is logged-in"
+    (let [user (user-db/create (factories/user))
+          response (handler/login-form {:user (select-keys user
+                                                           [:users/id
+                                                            :users/name
+                                                            :users/role
+                                                            :users/email])})]
+      (is (= 302 (:status response)))
+      (is (= "/users/dashboard"  (get-in response [:headers "Location"]))))))
+
+(deftest logout
+  (testing "Should logout user when the user is logged-in"
+    (let [_ (handler/create-user {:form-params {:name "Severus Snape"
+                                                :email "s.snape@hogwarts.edu"
+                                                :role "admin"
+                                                :password "lily"}})
+          user-id (:users/id (first (user-model/find-by-keys {:users/email "s.snape@hogwarts.edu"})))
```

Comment:
Why are we using handler to create a user in database? We could have used a factory(passing an encrypted password as an override), right?